### ISSUE TITLE
[Refactoring] Replace Xamarin.Firebase.iOS.CloudMessaging

### ISSUE THREAD

**Issue by thomasgalliker** [thomasgalliker/Plugin.FirebasePushNotifications#19]
## Summary
It looks like the referenced nuget Xamarin.Firebase.iOS.CloudMessaging is no longer maintained and must be replaced by an updated Firebase iOS binding library.

**Comment by gabsamples6** [comment]
Alternative 
https://github.com/Redth/DotNet.Platform.SlimBindings  that is what MS reccomends which might fit in with Notification has there isnt much binding

**Comment by mrhery** [comment]
Is there any sample codes to implement on the old app which has no MVVM model?

**Comment by thomasgalliker** [comment]
@mrhery This is not related to Xamarin iOS bindings. Please keep different problems in separate issues. For questions like this I would recommend to use the discussions section of this github repository. The answer is: No, there are no 1.x samples, because 1.x was a total mess and was just published because we had nothing better at the time. Update to 2.x and you will have a sample app. If anything is broken in 2.x or later, open a new issue.

**Comment by gaurav-bagga** [comment]
@thomasgalliker When can we expect this issue to be complete, as mentioned in referenced ticket from CrossGeek's original project, old iOS bindings are not maintained and will get redundant in June 2024 and as we are close to the end date any pointers to when can we expect this would really help.

**Comment by thomasgalliker** [comment]
Did Google mention that you no longer receive push notifications when we dont update these bindings? They just say we have to update this at one point but the change in June 2024 SHOULD only affect the send API in the backend.

**Comment by gaurav-bagga** [comment]
@thomasgalliker I not a seasoned Xamarin developer, I have an old Xamarin app which I am trying to upgrade to MAUI.Net. I was using https://github.com/CrossGeeks/FirebasePushNotificationPlugin project. I was searching for a new library and stumbled upon some links which pointed to project https://github.com/AdamEssenmacher/GoogleApisForiOSComponents similar to this ticket.

When you visit the above project it says in overview old 

> App Store compliance requires native SDK versions 10.24+, while the Xamarin packages are stuck on 8.10.

Now I am confused if I use a project using Xamarin.Firebase.iOS.* will it impact me when I submit the application to App Store. 

What you replied above to my query earlier sorry I could not follow, what I earlier meant was every other library you search for Firebase they are trying to use https://github.com/AdamEssenmacher/GoogleApisForiOSComponents.

As far as any notice from Google I did not get any when using old Xamarin app, I just wanted to upgrade before App Store and Play Store stop accepting it because of using obsolete libraries and SDK.

**Comment by thomasgalliker** [comment]
Yes, @gaurav-bagga we also have to go this way and replace the iOS bindings. As far as I understand google‘s info mail, you will kerp receiving push notifications even after June 2024. Correct me if I‘m wrong. 
So, the only thing missing here is some free time to do this change.

**Comment by AlleSchonWeg** [comment]
In my opinion the biggest benefit is that with the new bindings you can receive notifications in the iOS simulator. Makes debugging easier.

**Comment by thomasgalliker** [comment]
@AlleSchonWeg @gaurav-bagga @gabsamples6 @mrhery
Do you like to test the latest version 3.0.x? It contains the firebase ios sdk 10.27.

**Comment by thomasgalliker** [comment]
This change is released in version 3.0.17. Thanks everyone for the patience.
